src/components/forms/Search/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { History } from 'history';
+import React, { useState } from 'react';
+import { Button, Form, FormGroup, Input } from 'reactstrap';
+import { SEARCH } from '../../../configs/lang';
+
+/**
+ * Interface for handleSearchChange event handler
+ */
+export type Change = (event: React.ChangeEvent<HTMLInputElement>) => void;
+
+/**
+ * Interface for handleSubmit event handler
+ */
+export type Submit = (event: React.FormEvent<HTMLFormElement>) => void;
+
+/**
+ * Interface for SearchForm props
+ */
+export interface SearchFormProps {
+  history: History;
+  placeholder?: string;
+}
+
+/** Search Form component */
+export const SearchForm = (props: SearchFormProps) => {
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleSearchChange: Change = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+  };
+
+  const handleSubmit: Submit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    props.history.push({
+      search: `?search=${searchQuery}`,
+    });
```

Comment:
@kelvin-muchiri Do you think it would be a good Idea to only push a new url if the searchQuery is not an empty string, or maybe activate the search button only when there is some input.

Reply:
Since we reintroduced the search button, it is also used to clear the search

Reply:
Basically to clear the search results, clear the search input then submit.

Reply:
what I meant  was that you would have submit disabled untill we had some input in the text field, I do not think it collides with this.

> Basically to clear the search results, clear the search input then submit.

Anyway I do not have any strong feelings around this, so I gues its still fine, the way its is.

Reply:
@kelvin-muchiri @p-netm if we allow the user to submit with an empty string, could we just navigate to the current url without the search params (effectively just showing unfiltered data)?